web/src/compose_validate.ts

```diff
@@ -184,23 +201,28 @@ export function warn_if_private_stream_is_linked(linked_stream, $textarea) {
     }
 }
 
-export function warn_if_mentioning_unsubscribed_user(mentioned, $textarea) {
+export function warn_if_mentioning_unsubscribed_user(
+    mentioned: UserOrMention,
+    $textarea: JQuery<HTMLTextAreaElement>,
+): void {
     // Disable for Zephyr mirroring realms, since we never have subscriber lists there
     if (realm.realm_is_zephyr_mirror_realm) {
         return;
     }
+    let user_id: number;
 
-    const user_id = mentioned.user_id;
+    if ("user_id" in mentioned) {
+        user_id = mentioned.user_id;
+    } else {
+        return;
+    }
```

Comment:
This refactor seems nontrivial; can you talk about why it's correct?

Reply:
`UserOrMention` is a union of `PseudoMentionUser` and `User`. `user_id` is only present in `User`, so this cause an error and hence, the check.

Reply:
Thanks. I have corrected this.